src/main/java/org/zalando/nakadi/repository/kafka/KafkaTopicRepository.java

```diff
@@ -287,22 +289,51 @@ private void failUnpublished(final List<BatchItem> batch, final String reason) {
     @Override
     public Optional<PartitionStatistics> loadPartitionStatistics(final Timeline timeline, final String partition)
             throws ServiceUnavailableException {
+        return loadPartitionStatistics(Collections.singletonList(new TimelineAndPartition(timeline, partition))).get(0);
+    }
+
+    @Override
+    public List<Optional<PartitionStatistics>> loadPartitionStatistics(
```

Comment:
I'm a bit confused now, there is also method `public List<PartitionStatistics> loadTopicStatistics(final Collection<Timeline> timelines)` and they have a lot of similar code. Can this method `loadTopicStatistics` be implemented using this new method `loadPartitionStatistics` to avoid code duplication?